### PR TITLE
Use BOTH library ID and email if provided by non-SUNet users for requ…

### LINKED ISSUE #493: Users who enter invalid library ID are then not able to request by name + email
User enters a library ID plus name + email in a page request. Library ID is bad - either expired, or not the right kind, or whatever. Request fails. User then tries to enter only name + email, but we have permanently linked the name + email to the ID, and they continue to get "request failed" messages. User workaround is to use a different email address. Backend workaround is to manually delete the library ID from the requests database. Would be better to just use the info entered, and not link name + email to a previously entered ID.

SW-1588 plus December discussion in Slack channel around user marvina.

### PR DESCRIPTION
**Pull request by tingulfsen**
…ests.

Fixes #493

### CODE REVIEW

File: app/models/request.rb [hunk 1]
```diff
@@ -80,12 +80,20 @@ def find_existing_user
...
-      User.find_by_library_id(user.library_id)
+      find_exisisting_library_id_user
     elsif user.non_webauth_user?
       find_existing_email_user
     end
   end
 
+  def find_exisisting_library_id_user
+    if user.email
+      User.find_by('library_id = ? AND email = ?', user.library_id, user.email)
```
Comment (jkeck) [inline app/models/request.rb]: Active record provides methods to do this type of thing natively.

`User.where(library_id: user.library_id, email: user.email).first`

File: app/models/request.rb [hunk 2]
```diff
@@ -80,12 +80,20 @@ def find_existing_user
     if user.webauth_user?
       User.find_by_webauth(user.webauth)
     elsif user.library_id_user?
-      User.find_by_library_id(user.library_id)
+      find_exisisting_library_id_user
```
Comment (jkeck) [inline app/models/request.rb]: Is there a good way to test for this behavior?

Comment (ndushay) [inline app/models/request.rb]: `exisisting` ... typo

File: spec/models/request_spec.rb [hunk 3]
```diff
@@ -268,6 +268,22 @@
...
+          item_id: '1234',
+          origin: 'GREEN',
+          origin_location: 'STACKS',
+          user_attributes: {
+            library_id: '12345',
+            email: 'jstanford@stanford.edu'
+          }
+        )
+        expect(User.where(library_id: '12345', email: 'jstanford@stanford.edu').length).to eq 1
+      end
```
Comment (jkeck) [inline spec/models/request_spec.rb]: Hmm, I'm curious if this actually tests our bug.  Does this fail before your change?

**Comment by jkeck** [comment]
@tingulfsen and I paired on this a bit and think that we have a small change and tests that address the issue.  If anybody else wants to look over it before we merge, please feel free.

File: spec/models/request_spec.rb [hunk 4]
```diff
@@ -278,6 +278,43 @@
...
+
+      it 'should not duplicate user records when both library ID and email is provided' do
```
Comment (ndushay) [inline spec/models/request_spec.rb]: if this test is doing something different from the above test, this description is not making it clear - I am confused.

File: spec/models/request_spec.rb (hunk 3 above)
Comment (ndushay) [inline spec/models/request_spec.rb]: jkeck asks a good question - was it answered F2F?

**Pull request merged**